File: 0907-koko-eating-bananas/0907-koko-eating-bananas.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int n=piles.length;
        long low=1;
        long high=Integer.MIN_VALUE;
        // for(int i=0;i<n;i++){
        //     low=(long)Math.min(low,piles[i]);
        // }
        for(int i=0;i<n;i++){
            high=(long)Math.max(high,piles[i]);
        }
        long ans=0;
        while(low<=high){
            long k=low+(high-low)/2;
            if(solve(piles,h,k)){
                ans=k;
                high=k-1;
            }else{
                low=k+1;
            }
        }
        return (int)ans;
    }
    static boolean solve(int[] nums,long h,long k){
        long count=0;
        for(int i=0;i<nums.length;i++){
           long rem=nums[i]%k;
           if(rem>0)count = count + nums[i]/k +1;
           else count=count+nums[i]/k;
        }
        if(count<=h) return true;
        return false;
    }
    
}

File: 0907-koko-eating-bananas/0907-koko-eating-bananas.js
/**
 * @param {number[]} piles
 * @param {number} h
 * @return {number}
 */
var minEatingSpeed = function(piles, h) {
    const n=piles.length;
    let low=1;
    let high=-Infinity;
    for(let i=0;i<n;i++){
        high=Math.max(high,piles[i]);

    }
    let ans=0;
    while(low<=high){
        let mid= Math.floor(low+(high-low)/2);
        if(solve(piles,h,mid)){
            ans=mid;
            high=mid-1;
        }else{
            low=mid+1;
        }
    }
    return ans;
};
 function solve(nums,h,mid){
    let count = 0;
    for (let i = 0; i < nums.length; i++) {
        let rem = nums[i] % mid;
        if (rem > 0) count = count + Math.floor(nums[i] / mid) + 1;
        else count = count + Math.floor(nums[i] / mid);
    }
    return count <= h;
 }